packages/material-ui/src/CardHeader/CardHeader.spec.tsx

```diff
@@ -0,0 +1,180 @@
+  // $ExpectError
+  <CardHeader component="incorrectComponent" />;
+  // $ExpectError
+  <CardHeader component={CustomComponent} />;
+}
+
+function mixedCardHeaderComponentAndTypographyTest() {
+  <CardHeader component="div" titleTypographyProps={{ component: 'a', href: 'href' }} />;
+  <CardHeader component="div" subheaderTypographyProps={{ component: 'a', href: 'href' }} />;
+  <CardHeader
+    component={CustomComponent}
+    stringProp="string"
+    numberProp={1}
+    titleTypographyProps={{ component: CustomComponent, stringProp: 'stringProp', numberProp: 2 }}
+  />;
+  <CardHeader
+    component={CustomComponent}
+    stringProp="string"
+    numberProp={1}
+    titleTypographyProps={{ component: CustomComponent, stringProp: 'stringProp', numberProp: 2 }}
+    subheaderTypographyProps={{
+      component: CustomComponent,
+      stringProp: 'stringProp',
+      numberProp: 2,
+    }}
+  />;
+  // $ExpectError
+  <CardHeader component="incorrectComponent" />;
+  // $ExpectError
+  <CardHeader component={CustomComponent} />;
+  <CardHeader
+    component={CustomComponent}
+    stringProp="string"
+    numberProp={1}
+    // $ExpectError
+    titleTypographyProps={{ component: CustomComponent, stringProp: 'stringProp' }}
+    subheaderTypographyProps={{
+      component: CustomComponent,
+      stringProp: 'stringProp',
+      numberProp: 2,
+    }}
+  />;
+  // $ExpectError
+  <CardHeader
+    component={CustomComponent}
+    stringProp="string"
+    numberProp={1}
+    titleTypographyProps={{ component: CustomComponent, stringProp: 'stringProp' }}
+    subheaderTypographyProps={{ component: CustomComponent, stringProp: 'stringProp' }}
+  />;
+  <CardHeader
+    // $ExpectError
+    component="incorrectComponent"
+    stringProp="string"
+    numberProp={1}
+    titleTypographyProps={{ component: CustomComponent, stringProp: 'stringProp', numberProp: 2 }}
+    subheaderTypographyProps={{
+      component: CustomComponent,
+      stringProp: 'stringProp',
+      numberProp: 2,
+    }}
+  />;
+}
+
+function titleTypographyPropsTest() {
+  // $ExpectError
+  <CardHeader titleTypographyProps={{ component: 'incorrectComponent' }} />;
+  <CardHeader titleTypographyProps={{ component: 'a', href: 'href' }} />;
+  <CardHeader
+    titleTypographyProps={{ component: CustomComponent, stringProp: 'stringProp', numberProp: 2 }}
+  />;
+  <CardHeader titleTypographyProps={{ variant: 'h1' }} />;
+  <CardHeader titleTypographyProps={{ align: 'left' }} />;
+  <CardHeader
+    titleTypographyProps={{
+      color: 'primary',
+      display: 'block',
+      gutterBottom: true,
+      noWrap: true,
+      variantMapping: { h1: 'h1' },
+    }}
+  />;
+  // $ExpectError
+  <CardHeader titleTypographyProps={{ component: CustomComponent, numberProp: 2 }} />;
+  <CardHeader
+    titleTypographyProps={{
+      component: 'a',
+      // $ExpectError
+      propThatDoesntExist: 'shouldNotWork',
+    }}
+  />;
```

Comment:
@eps1lon @fyodore82 need some advice...

CardHeader has an additional level of complexity compared to `ListItemText` in that it's a type map.

The component prop on both `titleTypographyProps ` & `subheaderTypographyProps` now works as expected. However I can't seem to find the way to get these tests to fail. The nested typography prop objects accept any number of unknown props. The correct typography props work as expected and the optional component prop too but the typing system just let's any other unknown props go through as well.

1. Is this an issue? ListItemText rejects unknown props.
2. If it is an issue, any ideas on how to address?

Reply:
I don't have the bandwidth to deep-dive into this issue. Maybe some issue with to many free type parameters or some quirk in JSX.

For now just document that this is undesired behavior in code comments. In addition it might be nice to extend these undesired behavior with tests using `createElement` directly (instead of JSX) and if wrong value types are rejected e.g. { variant: 123213 } or { component: SomeComponentImplementingFooNumber, foo: 'a-string-which-should-be-a-number' }

Reply:
- Documented the unmet expectations via comments
- Added createElement tests
  - Behaviour is pretty consistent with JSX. Only exception I found was the typography props don't seem to require the required props from the `component` prop type like it does in JSX.
  - Wrong value types _are_ indeed rejected (e.g. `{ variant: 123213 }`, etc)